lean4-infoview/src/infoview/rpcSessions.tsx

```diff
@@ -244,9 +249,13 @@ export class RpcSessions implements Disposable {
         })
     }
 
+    ensureAllSessionsClosed() {
+        [...this.#connected.keys()].forEach(uri => this.ensureSessionClosed(uri));
+        [...this.#connecting.keys()].forEach(uri => this.ensureSessionClosed(uri));
+    }
+
     dispose() {
-        this.#connected.forEach(rs => rs.dispose())
-        this.#connecting.forEach(fut => fut.then(rs => rs?.dispose()))
+        this.ensureAllSessionsClosed();
     }
 }
 
```

Comment:
I don't understand line 256 the `React.useEffect` containing `return () => sessions.dispose()` can you explain this one to me?

Reply:
Never mind I read up on `useEffect` and see now that the empty dependencies `[]` cause this to be a `componentDidMount` event only and it is returning the function to be invoked on unmount, which is the sessions.dispose...

Reply:
Indeed, this cleanup handler will only be executed if we tell react to display something else.  It would be great if we could cleanly shutdown the RPC sessions when the infoview is closed, but I don't know if this is possible.  We are currently using a keepalive timer to clean up inactive sessions on the server side for this reason.